File: Examples/GroupDocs.Annotation.Examples.CSharp/GroupDocs.AnnotationDataStorage/Repositories/UserRepository.cs
using System;
using GroupDocs.Annotation.Handler.Input;
using GroupDocs.Annotation.Handler.Input.DataObjects;

namespace GroupDocs.Data.Json.Repositories
{
    public class UserRepository : JsonRepository<User>, IUserDataHandler
    {
        private const string _repoName = "GroupDocs.users.json";

        public UserRepository(IRepositoryPathFinder pathFinder)
            : base(pathFinder.Find(_repoName))
        {
        }
        public User GetUserByGuid(string guid)
        {
            lock (_syncRoot)
            {
                try
                {
                    return Data.Find(x => x.Guid == guid);
                }
                catch(Exception e)
                {
                    throw new DataJsonException("Failed to get user by GUID.", e);
                }
            }
        }

        public User GetUserByEmail(string email)
        {
            lock (_syncRoot)
            {
                try
                {
                    return Data.Find(x => x.Email == email);
                }
                catch(Exception e)
                {
                    throw new DataJsonException("Failed to get user by name.", e);
                }
            }
        }
    }
}

File: Examples/GroupDocs.Annotation.Examples.VisualBasic/GroupDocs.AnnotationDataStorage/Repositories/UserRepository.vb
Imports GroupDocs.Annotation.Handler.Input
Imports GroupDocs.Annotation.Handler.Input.DataObjects

Namespace GroupDocs.Data.Json.Repositories
    Public Class UserRepository
        Inherits JsonRepository(Of User)
        Implements IUserDataHandler
        Private Const _repoName As String = "GroupDocs.users.json"

        Public Sub New(pathFinder As IRepositoryPathFinder)
            MyBase.New(pathFinder.Find(_repoName))
        End Sub
        Public Function GetUserByGuid(guid As String) As User Implements IUserDataHandler.GetUserByGuid
            SyncLock _syncRoot
                Try
                    Return Data.Find(Function(x) x.Guid = guid)
                Catch e As Exception
                    Throw New DataJsonException("Failed to get user by GUID.", e)
                End Try
            End SyncLock
        End Function

        Public Function GetUserByEmail(email As String) As User Implements IUserDataHandler.GetUserByEmail
            SyncLock _syncRoot
                Try
                    Return Data.Find(Function(x) x.Email = email)
                Catch e As Exception
                    Throw New DataJsonException("Failed to get user by name.", e)
                End Try
            End SyncLock
        End Function
    End Class
End Namespace